File: ruby/rabbitmq/consumer.rb
require 'bunny'

connection = Bunny.new(:host => "localhost", :vhost => "/", :user => "guest", :password => "guest")

def callback(channel, queue)
  queue = channel.queue(queue)
  queue.subscribe(:auto_ack => true) do |delivery_info, properties, payload|
    puts "Received #{payload}, message properties are #{properties.inspect}"
  end
end

connection.start
channel = connection.create_channel
queueName = 'to-be-processed'
callback(channel, queueName)
connection.close

File: python/rabbitmq/consumer.py
from modules.rabbitmq.connector import RabbitMQ

def callback(self, channel, method, properties, body):
    print(f"Received {body}, message properties are {properties}")

amqp = RabbitMQ()
connection = amqp.connect('localhost', 5672, 'guest', 'guest')
channel =  amqp.create_channel(connection)

queue='some-queue'

amqp.consume(channel, queue, callback)